config.py

```diff
@@ -788,3 +770,106 @@ def default_reject_comps_factory():
 
 if use_ssp and use_ica:
     raise ValueError('Cannot use both SSP and ICA.')
+
+if not ch_types:
+    msg = 'Please specify ch_types in your configuration.'
+    raise ValueError(msg)
+
+if ch_types == ['eeg']:
+    pass
+elif 'eeg' in ch_types and len(ch_types) > 1:  # EEG + some other channel types
+    msg = ('EEG data can only be analyzed separately from other channel '
+           'types. Please adjust `ch_types` in your configuration.')
+    raise ValueError(msg)
+elif any([ch_type not in ('meg', 'mag', 'grad') for ch_type in ch_types]):
+    msg = ('Invalid channel type passed. Please adjust `ch_types` in your '
+           'configuration.')
+    raise ValueError(msg)
+
+if 'eeg' in ch_types:
+    if use_ssp:
+        msg = ('You requested SSP for EEG data via use_ssp=True. However, '
+               'this does not work. Please ICA instead by setting '
+               'use_ssp=False and use_ica=True.')
+        raise ValueError(msg)
+    if not use_ica:
+        msg = ('You did not request ICA artifact correction for your data. '
+               'To turn it on, set use_ica=True.')
+        mne.utils.warn(msg)
```

Comment:
a warn suggests that ICA is mandatory for EEG. This is not for me.  I would make it an info message.

Reply:
Changed to an ordinary log message in 0141852